src/controller/CHIPDeviceController.cpp

```diff
@@ -309,24 +309,57 @@ CHIP_ERROR DeviceController::ServiceEventSignal()
     return err;
 }
 
-void DeviceController::OnNewConnection(Transport::PeerConnectionState * peerConnection, SecureSessionMgrBase * mgr) {}
+void DeviceController::OnNewConnection(SecureSessionHandle session, SecureSessionMgrBase * mgr)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint16_t index = 0;
+
+    VerifyOrExit(mState == State::Initialized, err = CHIP_ERROR_INCORRECT_STATE);
+
+    index = FindDeviceIndex(mgr->GetPeerConnectionState(session)->GetPeerNodeId());
+    VerifyOrExit(index < kNumMaxActiveDevices, err = CHIP_ERROR_INVALID_DEVICE_DESCRIPTOR);
+
+    mActiveDevices[index].OnNewConnection(session, mgr);
```

Comment:
This makes the device object use the secure session corresponding to the keyID that was derived during pairing (which is based on SPAKE2P handshake). How will it switch to using the key derived by SIGMA handshake (which is more secure, and should typically be used for all communication once pairing process is complete).

Reply:
Maybe I'm not quite understand your question, but IMHO, `SecureSessionHandle` can represent both `SPAKE2` and `SIGMA` session. And there can be multiple connections to a device, `OnNewConnection` is triggered for every connection.

If the `Device` class wants to "persistent" a connection, it can choose one connection it wants.

Reply:
Sorry, let me rephrase the question. It's two parts.

1. What's the use case where an application decides to use a `PASE` based connection (established by SPAKE2P), after a `CASE` based connection (established via SIGMA) is already established?
2. How would the application know which KeyID corresponds to which connection. (it's ok if this logic is added in a follow on PR).

Reply:
I have created another PR to address the problem (https://github.com/project-chip/connectedhomeip/pull/4171), Let's continue the discussion on that PR.